src/main/java/seedu/duke/Account.java

```diff
@@ -28,10 +29,14 @@ public Currency getCurrencyType () {
      * @throws NotEnoughInAccountException if the balance would become negative
      */
     public void updateBalance (float changeInBalance, String action) throws NotEnoughInAccountException,
-            InvalidUpdateBalanceActionException {
+            InvalidUpdateBalanceActionException, TooLargeAmountException {
         int newBalance;
         if (action.equals("add")) {
             newBalance = balance + (int) (changeInBalance * 100);
+            final int UPPER_BOUND = 1_000_000_000;
+            if (newBalance > UPPER_BOUND) {
+                throw new TooLargeAmountException();
+            }
```

Comment:
I realise that if set the upper boundary to 1,000,000,00 it will still break. I think because `int` has max size of  2147483647
![image](https://user-images.githubusercontent.com/56477330/225334057-760b6bed-d7e4-4b17-9f7f-dc0d2052e71b.png)